generics/services/sunbird.js

```diff
@@ -0,0 +1,239 @@
+*/
+
+const createUser = async function (userInputData, token) {
+    return new Promise(async (resolve, reject) => {
+
+        const createUserUrl =
+        process.env.SUNBIRD_URL + constants.apiEndpoints.SUNBIRD_CREATE_USER;
+
+
+        let createUser = {
+            firstName: userInputData.firstName,
+            lastName: userInputData.lastName,
+            phoneNumber: userInputData.phoneNumber,
+            userName: userInputData.userName,
+            password: userInputData.password,
+            gender: userInputData.gender ? userInputData.gender.value : ""
+
+        }
+
+        if (userInputData.dateOfBirth) {
+            createUser['dob'] = userInputData.dateOfBirth;
+        }
+
+        if (userInputData.email) {
+            createUser['email'] = userInputData.email;
+            createUser['emailVerified'] = true;
+        }
+        if (userInputData.phoneNumber) {
+            createUser['phone'] = userInputData.phoneNumber;
+            createUser['phoneVerified'] = true;
+        }
+        if (userInputData.address) {
+            createUser['address'] = [{
+                "addressLine1": address,
+                "city": address
+            }]
+        }
+
+        let response = await callToSunbird(token, createUser, createUserUrl);
+        return resolve(response);
+    })
+
+}
+
+
+/**
+  * add user to organisation
+  * @function
+  * @name addUserToOrganisation
+  * @param requestBody - body data for creating user.
+  * @param token - Logged in user token.
+  * @returns {json} response consist of sunbird api response details 
+*/
+
+const addUserToOrganisation = async function (requestBody, token) {
+    return new Promise(async (resolve, reject) => {
+
+        const adduserToOrgUrl =
+        process.env.SUNBIRD_URL + constants.apiEndpoints.SUNBIRD_ADD_USER_TO_ORG;
+
+        let response = await callToSunbird(token, requestBody, adduserToOrgUrl);
+        return resolve(response);
+    })
+
+}
+
+
+
+/**
+  * update user
+  * @function
+  * @name updateUser
+  * @param requestBody - body data for update user.
+  * @param token - Logged in user token.
+  * @returns {json} - Response consists of updated user details
+*/
+
+const updateUser = async function (requestBody, token) {
+    return new Promise(async (resolve, reject) => {
+
+        const updateUserApiUrl =
+            process.env.SUNBIRD_URL + constants.apiEndpoints.SUNBIRD_UPDATE_USER;
+
+        let response = await callToSunbird(token, requestBody, updateUserApiUrl, "PATCH");
+        return resolve(response);
+
+    })
+
+}
+
+
+/**
+  * Call to sunbird api's. 
+  * @function
+  * @name callToSunbird
+  * @param requestBody - Logged in user Id.
+  * @param token - Logged in user token.
+  * @param url - url of the api call.
+  * @param requestType - http request method
+  * @returns {JSON} - user profile information.
+*/
+
+function callToSunbird(token, requestBody, url, requestType = "") {
```

Comment:
@rakeshSgr - I would suggest having the arguments in the sequence as - requestType, url, token and req body so that no need to pass the last request body for get call if not required.

Reply:
resolved